src/github.com/getlantern/lantern-android/README.md

```diff
@@ -0,0 +1,232 @@
+* [Go 1.4][4]
+* [GNUMake][6]
+
+### Setting up a development environment
+
+We're going to clone and use the [flashlight-build][5] repository, that
+project provides us with everything we need to build Lantern tools and
+libraries.
+
+```sh
+mkdir -p $GOPATH/src/github.com/getlantern
+cd $GOPATH/src/github.com/getlantern
+git clone https://github.com/getlantern/flashlight-build.git
+```
+
+After cloning the repository use the `setenv.bash` script to export development
+enviromental variables to the current terminal session.
+
+```
+cd flashlight-build
+source setenv.bash
+```
+
+## Building the Android library
+
+`flashlight-build` will provide you with a copy of `lantern-android`, change
+directory into `lantern-android` and execute `make` to start compiling our
+shared library:
+
+```
+cd $GOPATH/src/github.com/getlantern/lantern-android
+
+make
+# ...
+# BUILD SUCCESSFUL
+# Total time: 7 seconds
+```
+
+The `make` command will create a new `app` subdirectory that will contain an
+Android example project. You may import the contents of the `app` subdirectory
+into Android Studio to see libflashlight working.
+
+## Testing the example project
+
+Open [Android Studio][3] and in the welcome screen choose "Import Non-Android
+Studio project".
+
+![Android Studio](https://cloud.githubusercontent.com/assets/385670/5712830/5f4cda3c-9a7b-11e4-85af-8af9d54e18c7.png)
+
+You'll be prompted with a file dialog, browse to the `app` subdirectory and
+select it. Press *OK*.
+
+![App Subdirectory](https://cloud.githubusercontent.com/assets/385670/5769230/5431dec6-9cde-11e4-82ce-d3983471a1f1.png)
+
+On the next dialog you must define a destination for the project, hit *Next*.
+
+![Destination](https://cloud.githubusercontent.com/assets/385670/5712874/ad8265e6-9a7b-11e4-9018-671875dfdb17.png)
+
+After import you may be prompted to restart Android Studio. After restarting
+make sure your `AndroidManifest.xml` looks like this:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+Copyright 2014 The Go Authors. All rights reserved.
+Use of this source code is governed by a BSD-style
+license that can be found in the LICENSE file.
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="org.getlantern.example" android:versionCode="1" android:versionName="1.0">
+
+  <application android:label="Flashlight">
+    <activity android:name="org.getlantern.example.MainActivity"
+      android:label="Flashlight"
+      android:exported="true">
+      <intent-filter>
+        <action android:name="android.intent.action.MAIN" />
+        <category android:name="android.intent.category.LAUNCHER" />
+      </intent-filter>
+    </activity>
+  </application>
+  <uses-permission android:name="android.permission.INTERNET" />
+</manifest>
+```
+
+Now add a new *main activity* by right-clicking on the top most directory on
+the *Project* pane and selecting New->Activity->Blank Activity, the default
+values would be OK, click *Finish*.
```

Comment:
Weird, mine is older than yours, even though I just updated it ...


![screen shot 2015-01-27 at 5 54 38 pm](https://cloud.githubusercontent.com/assets/5000654/5929831/bb608052-a64d-11e4-9370-32cf063e4587.png)